spec/unit/agent_spec.rb

```diff
@@ -85,12 +83,6 @@ def controlled_run(&block)
       @agent.run
     end
 
-    it "should do nothing if already running" do
-      @agent.expects(:running?).returns true
-      AgentTestClient.expects(:new).never
-      @agent.run
-    end
-
```

Comment:
Should we check if the lock raises `Puppet::LockError` then we expect `client.expects(:run).never` and a notice to be logged, e.g. `Puppet.expects(:notice).with(regexp_matches(/not sure what goes here../))`